src/components/dialogs/parameters/dynamicsimulation/curve/dialog/model-filter.js

```diff
@@ -91,6 +96,31 @@ const variablesTreeToVariablesArray = (variablesTree, parentId) => {
     return result;
 };
 
+const makeGetModelLabel = (intl) => (value) =>
+    intl.formatMessage({
+        id: `${PREFIX_MODEL_KEY}/${value}`,
+    });
+
+const makeGetVariableLabel = (intl) => (elem) => {
+    if (!elem.parentId) {
+        // root element => that is model element in the variable tree
+        return intl.formatMessage({ id: `${PREFIX_MODEL_KEY}/${elem.name}` });
+    } else {
+        // either a variable set element or variable element in the variable tree
+        if (elem.variableId) {
+            // that is a variable element
+            return intl.formatMessage({
+                id: `${PREFIX_VARIABLE_KEY}/${elem.name}`,
+            });
+        } else {
+            // must be a variable set element
+            return intl.formatMessage({
+                id: `${PREFIX_VARIABLE_SET_KEY}/${elem.name}`,
+            });
+        }
+    }
+};
```

Comment:
For this type of function, I think it is easier to read if we do not write "else" after a return, like this : 
```suggestion
const makeGetVariableLabel = (intl) => (elem) => {
    if (!elem.parentId) {
        // root element => that is model element in the variable tree
        return intl.formatMessage({ id: `${PREFIX_MODEL_KEY}/${elem.name}` });
    } 
    // either a variable set element or variable element in the variable tree
    if (elem.variableId) {
        // that is a variable element
        return intl.formatMessage({
            id: `${PREFIX_VARIABLE_KEY}/${elem.name}`,
        });
    } 
    // must be a variable set element
    return intl.formatMessage({
        id: `${PREFIX_VARIABLE_SET_KEY}/${elem.name}`,
    });
};
```

Reply:
Yes! DONE